CHNS example : capillary rise and Jurin's law (#1328)
chaos-polymtl/lethe: doc/source/examples/multiphysics/jurins-law/jurins-law.rst

|     :alt: Computational domain definition and example of a capillary rise for an arbitrary (<90°) angle of contact|
|     :align: center                                                                                                |
|     :name: Computational domain and example of a capillary rise for an arbitrary (<90°) angle of contact          |
|                                                                                                                   |
|     Representation of the initial (left) and final (right) states of the capillary rise                           |
|                                                                                                                   |
+-------------------------------------------------------------------------------------------------------------------+

The quantity of interest of this problem is the difference in height (:math:`\Delta H` in the previous figure) between the tip of the meniscus and the surface of the fluid outside of the central part. The Jurin's law [#liu2018]_ gives an asymptotical value of :math:`\Delta H` :

.. math::
    \Delta H = \frac{\sigma\cos{\alpha_c}}{\rho_1gR}

> with :math:`\sigma` the surface tension coefficient, :math:`\alpha_c` the angle of contact imposed on the wall, :math:`\rho_1` the density of the denser fluid, :math:`g` the gravitational acceleration and :math:`R` the radius of the central part.

Author: Good catch, I kept the name radius but half-distance fits better for the problem under study

-----------------
Parameter File
-----------------

Simulation Control
~~~~~~~~~~~~~~~~~~

Time integration is handled by a 1st order backward differentiation scheme (`bdf1`), for a :math:`0.5 \ \text{s}` simulation time with an initial time step of :math:`0.0005 \ \text{s}`. Time-step adaptation is enabled using ``adapt=true`` and the max CFL is :math:`0.8`. ``output boundaries`` is set to ``true`` to get a ``.vtu`` file containing the indices of the boundaries of the domain.

.. code-block:: text

    subsection simulation control
      set method            = bdf1
      set output name       = jurins-law-2d
      set output frequency  = 10
      set output path       = ./
      set max time step     = 5e-4
      set adapt             = true
      set max cfl           = 0.8
      set time end          = 0.5
      set time step         = 5e-4
      set output boundaries = true

>     end

Author: It should be 0.5 everywhere

        set density                    = 1
      end
        subsection fluid 1
        set kinematic viscosity        = 1e-4
        set density                    = 2000
      end
      set number of material interactions = 1
      subsection material interaction 0
        subsection fluid-fluid interaction
          set surface tension coefficient     = 7.3e-2
          set cahn hilliard mobility model    = constant
          set cahn hilliard mobility constant = 1e-7
        end
      end
    end

Cahn-Hilliard
~~~~~~~~~~~~~

In the ``cahn hilliard`` subsection, we set the ``potential smoothing coefficient`` (soon to be deprecated) to :math:`0`. The interface thickness is set to be determined automatically based on the mesh size in the ``epsilon`` subsection. We also output the interface thickness for each time-step by setting the ``verbosity`` to ``verbose`` to know its exact value for the initial conditions.

.. code-block:: text

    subsection cahn hilliard
      set potential smoothing coefficient = 0

      subsection epsilon
        set method    = automatic
        set verbosity = verbose
      end
    end

Initial Conditions
~~~~~~~~~~~~~~~~~~

> In the ``initial conditions`` subsection, we need only need to initialize the phase field in the ``cahn hilliard`` subsection. The chemical potential field is set to :math:`0` uniformly. The interface is initialized with the equilibrium interface thickness, which requires to know the value of :math:`\epsilon` that is outputed at every iteration.

Reviewer: ```suggestion
In the ``initial conditions`` subsection, we need only need to initialize the phase field in the ``cahn hilliard`` subsection. The chemical potential field is set to :math:`0` uniformly. The interface is initialized with the equilibrium interface thickness, which requires to know the value of :math:`\epsilon` that is outputted at every iteration. Here, :math:`\epsilon = 0.04419`.
```

    set number = 1
    	subsection bc 0
    		set id          = 2 
    		set type        = angle_of_contact
    		set angle value = 50
    	end
    end

Then, a ``slip`` boundary condition is applied everywhere, except for the upper boundary, where it is set as ``none``.

.. code-block:: text

    subsection boundary conditions
      set number = 4
      subsection bc 0
        set id   = 2 
        set type = slip
      end
      subsection bc 1
        set id   = 5 # walls
        set type = slip
      end
      subsection bc 2
        set id = 4 # upper surface
        set type = none
      end
      subsection bc 3
        set id   = 3 # middle
        set type = slip
      end
    end

Source Term
~~~~~~~~~~~

In the ``source term`` subsection, we define the gravitational acceleration. Since the unit length is the millimeter, the usual value of :math:`g` needs to be multiplied by :math:`1000`.

.. code-block:: text

    subsection source term
      subsection fluid dynamics
        set Function expression = 0; 0; -9810; 0
      end
    end

-----------------------
Running the Simulation
-----------------------

We call lethe-fluid by invoking:

> ``mpirun -np $number_of_CPU_cores lethe-fluid jurins-law-2d.prm``

Author: Very cool, I didn't know about this feature

|      pressure gradient at the vicinity of the interface corresponds to that expected by the Young-Laplace         |
|       equation,                                                                                                   |
|      with an overpressure at positive curvature interfaces and depressions at negative curvature interfaces.      |
|     :align: center                                                                                                |
|     :name: Pressure field at the end of the simulation                                                            |
|                                                                                                                   |
|     Pressure fields at the end of the simulation for different angles of contact.                                 |
|                                                                                                                   |
+-------------------------------------------------------------------------------------------------------------------+

---------------------------
Possibilities for Extension
---------------------------

> - **Going 3D**: the mesh can be extruded into the third dimension and there is an adaptation of the Jurin's law in three dimensions. Some results are available in the literature for comparison (see Lovrić et al. [#lovric2019]_)

Reviewer: ```suggestion
- **Going 3D**: the mesh can be extruded into the third dimension and there is an adaptation of the Jurin's law in three dimensions. Some results are available in the literature for comparison (see Lovrić *et al.* [#lovric2019]_)
```

> - **Investigate the effect of a no-slip boundary condition**: instead of the slip boundary condition imposed on the inner face of the wall, we could try to use a no-slip boundary condition. This situation would be closer to a real capillary rise experiment. We expect to observe a different transitory state with this new boundary condition.

Reviewer: ```suggestion
- **Investigate the effect of a no-slip boundary condition**: instead of the slip boundary condition imposed on the inner face of the wall, we could try to use a no-slip boundary condition. This situation would be closer to a real capillary rise experiment. We expect to observe a different transient state with this new boundary condition.
```
